typegate/src/runtimes/typegraph.ts

```diff
@@ -158,11 +170,29 @@ export class TypeGraphRuntime extends Runtime {
         };
 
         visitTypes(this.tg, getChildTypes(this.tg.types[0]), myVisitor);
-
-        const scalarTypes = distinctBy(
-          [...scalarTypeIndices].map((idx) => this.tg.types[idx]),
+        const distinctScalars = distinctBy(
+          [...scalarTypeIndices].map((idx) => {
+            const node = this.tg.types[idx];
+            this.scalarIndex.set(node.type, idx);
```

Comment:
this.customScalars looks to be only used locally, is there any need to have it class based?

Reply:
It's used in `formatType`

Reply:
`scalarIndex` seems to be local to this function or my browser search is buggy, `customScalars` is used in `formatType`

Reply:
yes, that is the case.

Reply:
`scalarIndex` will hold the number id of the scalar types as the name suggests, it is used in `formatType` and `getSchema`, then cleared after. I also removed the `customScalars` class attribute, items will be regenerated as we need them.